atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/HostPartitioner.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Multimaps;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+
+class HostPartitioner<V> {
+    private final CassandraClientPool cassandraClientPool;
+
+    HostPartitioner(CassandraClientPool cassandraClientPool) {
```

Comment:
nit: looks like these methods could be static instead - only one uses this variable and it could be passed in as an argument.